core/trino-main/src/main/java/io/trino/sql/planner/optimizations/ReplicateSemiJoinInDelete.java

```diff
@@ -72,13 +63,7 @@ public PlanNode visitDelete(DeleteNode node, RewriteContext<Void> context)
             // For delete queries, the TableScan node that corresponds to the table being deleted must be collocated with the Delete node,
             // so you can't do a distributed semi-join
             isDeleteQuery = true;
-            PlanNode rewrittenSource = context.rewrite(node.getSource());
```

Comment:
nit: it would be better to keep this line intact (in `Use PlanNode#replaceChildren for creating new PlanNode explicitly`)